pkg/supportbundle/spec.go

```diff
@@ -106,106 +106,200 @@ func CreateRenderedSpec(app apptypes.AppType, sequence int64, kotsKinds *kotsuti
-		return nil, errors.Wrap(err, "failed to encode support bundle")
-	}
-	renderedSpec = b.Bytes()
+		existingSecret, err := clientset.CoreV1().Secrets(util.PodNamespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+		labels := kotstypes.MergeLabels(kotstypes.GetKotsadmLabels(), kotstypes.GetTroubleshootLabels())
+		if err != nil {
+			if kuberneteserrors.IsNotFound(err) {
+				secret := &corev1.Secret{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "v1",
+						Kind:       "Secret",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      secretName,
+						Namespace: util.PodNamespace,
+						Labels:    labels,
+					},
+					Data: map[string][]byte{
+						SpecDataKey: renderedSpec,
+					},
+				}
 
-	secretName := GetSpecSecretName(app.GetSlug())
-	existingSecret, err := clientset.CoreV1().Secrets(util.PodNamespace).Get(context.TODO(), secretName, metav1.GetOptions{})
-	labels := kotstypes.MergeLabels(kotstypes.GetKotsadmLabels(), kotstypes.GetTroubleshootLabels())
-	if err != nil {
-		if kuberneteserrors.IsNotFound(err) {
-			secret := &corev1.Secret{
-				TypeMeta: metav1.TypeMeta{
-					APIVersion: "v1",
-					Kind:       "Secret",
-				},
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      secretName,
-					Namespace: util.PodNamespace,
-					Labels:    labels,
-				},
-				Data: map[string][]byte{
-					SpecDataKey: renderedSpec,
-				},
+				_, err = clientset.CoreV1().Secrets(util.PodNamespace).Create(context.TODO(), secret, metav1.CreateOptions{})
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to create support bundle secret")
+				}
+
+				logger.Debugf("created %q default support bundle spec secret", app.GetSlug())
+				// return supportBundle, nil
+			} else {
+				return nil, errors.Wrap(err, "failed to read support bundle secret")
 			}
+		} else {
+			if existingSecret.Data == nil {
+				existingSecret.Data = map[string][]byte{}
+			}
+			existingSecret.Data[SpecDataKey] = renderedSpec
+			existingSecret.ObjectMeta.Labels = labels
 
-			_, err = clientset.CoreV1().Secrets(util.PodNamespace).Create(context.TODO(), secret, metav1.CreateOptions{})
+			_, err = clientset.CoreV1().Secrets(util.PodNamespace).Update(context.TODO(), existingSecret, metav1.UpdateOptions{})
 			if err != nil {
-				return nil, errors.Wrap(err, "failed to create support bundle secret")
+				return nil, errors.Wrap(err, "failed to update support bundle secret")
 			}
-
-			logger.Debugf("created %q default support bundle spec secret", app.GetSlug())
-			return supportBundle, nil
-		} else {
-			return nil, errors.Wrap(err, "failed to read support bundle secret")
 		}
 	}
 
-	if existingSecret.Data == nil {
-		existingSecret.Data = map[string][]byte{}
+	// Include discovered support bundle specs and default kotsadm support bundle spec. Perform this action here so
+	// as not to add discovered specs to the default support bundle spec secret.c
+	return addDiscoveredSpecs(builtBundles["kotsadm"], app, clientset), nil
```

Comment:
Can we use a constant for this key? Could do this for the split bundle keys as well, but this one makes the most sense since we use it in a few places.

Reply:
Agree, I have created those constants of those keys in [pkg/kotsadm/types/constants.go](https://github.com/replicatedhq/kots/pull/3674/files#diff-6e08d80aec35b42dd85376cf4f15fa335bcd4f4a5438e436432d9d0a0b460278)